### ISSUE TITLE
Consecutive calls with expectation about args

### ISSUE THREAD

**Issue by mir3z** [sinonjs/sinon#1193]
I want to create a mock with the following expectations:

* it is called twice
* on first call it is called with args \<X\> and returns \<R1\>
* on second call it is called with args \<Y\> and returns \<R2\>

When I'm trying to do:
```
const mock = mock().twice();
mock.onCall(0).withArgs(0);
mock.onCall(1).withArgs(1);
```

I see error `Defining a stub by invoking "stub.onCall(...).withArgs(...)" is not supported. Use "stub.withArgs(...).onCall(...)" to define sequential behavior for calls with certain arguments.`

but if I follow the advice I'm unable to set different args and response for consecutive calls.

**Comment by mroderick** [comment]
The `withArgs` implementation in `sinon.mock` overwrites the stored value on each call. This is unfortunate, and should be improved.

Related: https://github.com/sinonjs/sinon/issues/992
Call for documentation: https://github.com/sinonjs/sinon/issues/1432

We would welcome PRs to improve this :)

**Comment by mroderick** [comment]
I am closing this as duplicate, since #992 reported it earlier.

**Comment by tkissing-work** [comment]
So you closed this as a dupe of #992 but then you closed that because now the behavior is documented?
That does not actually solve the problem :(

**Comment by pride7754** [comment]
@mir3z Have you found solution for this? I am also trying to do the same thing over here
